app/src/test/java/org/mozilla/tv/firefox/TestFirefoxApplication.kt

```diff
@@ -17,4 +20,12 @@ class TestFirefoxApplication : FirefoxApplication() {
      */
     override fun getSystemUserAgent(): String =
             "Mozilla/5.0 (Linux; Android 7.1.2) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/2.2.0.2 Chrome/59.0.3071.125 Mobile Safari/537.36"
+
+    /**
+     * This is used to disable ping upload when running tests.
+     */
+    override fun setGleanUpload() {
+        WorkManagerTestInitHelper.initializeTestWorkManager(applicationContext)
```

Comment:
I guess WorkManager is closely tied enough to the Glean work that it needs to be initialized, regardless of whether it's being used?

Reply:
Yes! We use the `WorkManager` internally for a few things, including the ping upload and metrics ping scheduling. The metrics ping scheduler doesn't get disabled when upload is disabled (for a series of reason, see [the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1552114) on our end), that's why the weird WorkManager init is required even if upload is disabled, in tests.